030_Data/45_Partial_update.asciidoc

```diff
@@ -86,50 +78,41 @@ POST /website/blog/1/_update
-including search, sorting, aggregations, and document updates. Scripts can be passed in as part of the request,
-retrieved from the special .scripts index, or loaded from disk.
+对于那些((("documents", "partial updates", "using scripts")))((("updating documents", "partial updates", "using scripts"))) API 不能满足需求的情况，Elasticsearch 允许你使用脚本编写自定义的逻辑。
+((("scripts", "using to make partial updates")))许多API都支持脚本的使用，包括搜索、排序、聚合和文档更新。
+脚本可以作为请求的一部分被传递，从特殊的 .scripts 索引中检索，或者从磁盘加载脚本。
+
+
+默认的脚本语言 ((("Groovy")))是 http://groovy.codehaus.org/[Groovy]，一种快速表达的脚本语言，在语法 JavaScript 类似。
+它在 Elasticsearch V1.3.0 版本首次引入并运行在 _沙盒_ 中，然而 Groovy 脚本引擎存在漏洞，
+允许攻击者通过构建 Groovy 脚本，在 Elasticsearch Java VM 运行时脱离沙盒并执行 shell 命令。
+
 
-The default scripting language ((("Groovy")))is http://groovy.codehaus.org/[Groovy], a
-fast and expressive scripting language, similar in syntax to JavaScript. It was first introduced
-in Elasticsearch version v1.3.0 and it runs in a _sandbox_, however there is vulnerability
-in the Groovy scripting engine that allows an attacker to construct
-Groovy scripts that escape the sandbox and execute shell commands as the user
-running the Elasticsearch Java VM.
+因此，在版本 v1.3.8 、 1.4.3 和 V1.5.0 及更高的版本中，它已经被默认禁用。
+此外，您可以通过设置集群中的所有节点的 `config/elasticsearch.yml` 文件来禁用动态 Groovy 脚本：
 
-Therefore in versions v1.3.8, v1.4.3, and version v1.5.0 and newer it has been disabled by default.
-Alternatively you can disable dynamic Groovy scripts by
-adding this setting to the `config/elasticsearch.yml` file in all nodes in the
-cluster:
 
 [source,yaml]
 -----------------------------------
 script.groovy.sandbox.enabled: false
 -----------------------------------
 
-This will turn off the Groovy sandbox, thus preventing dynamic Groovy scripts
-from being accepted as part of a request or retrieved from the special
-`.scripts` index. You will still be able to use Groovy scripts stored in files
-in the `config/scripts/` directory on every node.
+这将关闭 Groovy 沙盒，从而防止动态 Groovy 脚本作为请求的一部分被接受，
+或者从特殊的 `.scripts` 索引中被检索。当然，你仍然可以使用存储在每个节点的 `config/scripts/` 目录下的 Groovy 脚本。
 
-If your architecture and security is one that does not need worry about the vulnerability,
-for example your Elasticsearch endpoints are only exposed and available to trusted applications,
-then you can choose to re-enable the dynamic scripting if it is a feature your application needs.
 
-You can read more about scripting in the
-{ref}/modules-scripting.html[scripting reference documentation].
+如果你的架构和安全性不需要担心漏洞攻击，例如你的 Elasticsearch 终端仅暴露和提供给可信赖的应用，
+当它是你的应用需要的特性时时，你可以选择重新启用动态脚本。
+
+你可以在 {ref}/modules-scripting.html[scripting reference documentation] 获取更多关于脚本的资料.
 
 ****
 
-We can also use a script to add a new tag to the `tags` array.  In this
-example we specify the new tag as a parameter rather than hardcoding it in
-the script itself. This allows Elasticsearch to reuse the script in the
-future, without having to compile a new script every time we want to add
-another tag:
```

Comment:
We can also use a script to add a new tag to the `tags` array.  In this　example we specify the new tag as a parameter rather than hardcoding it in　the script itself.
这里是不是漏译了，我翻译了一下供参考：
我们也可以通过使用脚本给`tags`数组增加一个新的标签。在这个例子中，我们指定为新的标签指定一个变量参数代替字符编码在脚本中使用。